README.md
# MTADayZ
# Repository moved to https://github.com/ciber96/mtadayz


changelog.txt
# CHANGELOG #
This changelog lists all changes made to MTA DayZ. It adheres to the following features description.

## [VERSION] - DATE OF RELEASE (YYYY-MM-DD)
## Please add your change on rollingchangelog.txt.

## [0.9.8a PRE-RELEASE] - 2016-07-06 (YOU NEED REMOVE YOUR INTERNAL.DB FOR THIS UPDATE)
### NEW: Advanced Clothes System (+60 clothes)
### NEW: Beard and Hair System
### NEW: Weight System
### NEW: Interior System (with 4 interiors for the moment)
### ADDED: Add variable "weight_food" on "CfgServer.lua"
### ADDED: Add variable "weight_thirst" on "CfgServer.lua"
### ADDED: Add variable "weight_loose" on "CfgServer.lua"
### ADDED: Add variable "weight_loosetimer" on "CfgServer.lua"
### FIXED: body temperature keeps rising when unconscious bug
### DEPRECATED: Upgraded showPlayerHudComponent [deprecated] to setPlayerHudComponentVisible


# CLOTHES #

### ADDED: Military collar
### ADDED: Africa collar
### ADDED: Gold collar
### ADDED: Silver collar
### ADDED: LS collar
### ADDED: Black Bandana (M)
### ADDED: Blue Bandana (M)
### ADDED: Green Bandana (M)
### ADDED: Red Bandana (M)
### ADDED: Black Bandana (H)
### ADDED: Blue Bandana (H)
### ADDED: Green Bandana (H)
### ADDED: Red Bandana (H)
### ADDED: Black Beret
### ADDED: Red Beret
### ADDED: Old Hat
### ADDED: Black Hat
### ADDED: Yellow Hat
### ADDED: Black Trucker
### ADDED: Blue Trucker
### ADDED: Green Trucker
### ADDED: Red Trucker
### ADDED: Yellow Trucker
### ADDED: Cow-Boy Hat
### ADDED: White Hat
### ADDED: Hockey Mask
### ADDED: Black Shoe
### ADDED: Sport Shoe
### ADDED: Brown Shoe
### ADDED: Biker Shoe
### ADDED: Blue Shoe
### ADDED: Red Shoe
### ADDED: Beach Shoe
### ADDED: Black Pants
### ADDED: Beige Pants
### ADDED: Gray Shorts
### ADDED: Blue Shorts
### ADDED: Blue Jeans
### ADDED: Green Jeans
### ADDED: Gray Pants
### ADDED: Yellow Pants
### ADDED: Blue Jogging
### ADDED: Gray Jogging
### ADDED: Military Pants
### ADDED: Beige Vest
### ADDED: Baseball Shirt
### ADDED: Baseball 2 Shirt
### ADDED: Red Vest
### ADDED: Grey Shirt
### ADDED: Green Vest
### ADDED: Hawai Shirt
### ADDED: Black Vest
### ADDED: Brown Vest
### ADDED: Biker Vest
### ADDED: Blue Shirt
### ADDED: Green 2 Vest
### ADDED: Number 5 Shirt
### ADDED: Monk Shirt
### ADDED: Dark Glasses
### ADDED: Square Glasses
### ADDED: Red Glasses

Known Issue :
- The player can glitch during unconscious
- When a player has 2 same clothes in her inventory. If he drop one, the garment disappear on him (He needs to re-equip the garment).

## [0.9.7a] - DATE OF RELEASE (2016-06-24) (Last update : 2016-07-02)

### ADDED: For new features
### ADDED: Addon "Antiglitch"
### ADDED: Polish Language
### ADDED: Nice MTA DayZ DX to Login
### ADDED: Arma II chatbox style
### ADDED: Server Console Output on Global Chat
### ADDED: "Dead Night" Addon
### ADDED: Hide Body now available
### ADDED : gameplayVariables["waterdamage"] in cfgClient.lua, damage received by a player when it is under water
### ADDED: Anti-teleport detection method to BattlDayZ
### CHANGED: For changes in existing features
### CHANGED: Added Flashlight to Spanish Language
### CHANGED: Moved warning ([PLAYERNAME]'s Account was buggy and has been reset.) to side chat
### CHANGED: Pain now uses setCameraShakeLevel(150) & Added config gameplayVariables["painshakelevel"]
### CHANGED: New animation binds (1 = Salute, 2 = Hands up, 3 = Sit, 5 = Journal)
### CHANGED: playerThirst and playerHunger calculation every 30 seconds instead of onClientRender
### CHANGED: The night is far less dark
### CHANGED: The timer "night_check" is now set to 1 seconds
### CHANGED: You can now enable or disable the night in the "cfgClient.lua" file.
### DEPRECATED: For once-stable removed in upcoming releases
### REMOVED: For deprecated features removed in the release it's mentioned in
### FIXED: For all bugs
### FIXED: FPS Drop on some PC.
### FIXED: Anti-teleport detection
### FIXED: Fixed Polish Language Typo
### FIXED: Ammo bug
### FIXED: Patrol Pack bug
### FIXED: Black & White shader not loading
### FIXED: The player don't get damage if are submerged for along time
### FIXED: "NV Goggles" and "IR Goggles" do not work properly at night
### FIXED: Disappearance of the body after the death of a player
### SECURITY: To invite users to upgrade in case of vulnerabilities
### LANGUAGE: Romanian is now available.

# - #
## [0.9.5a] - 2015-08-19
### ADDED: New status effects: Sepsis, Infection & Unconsciousness
### ADDED: New Items: Range Finder, Epi-Pen
### ADDED: Crosshair integration
### ADDED: Difficulty options (Normal, Veteran & Hardcore)
### ADDED: Viral zombie (24000 blood, greenish skin)
### ADDED: Empty Gas Canister to industrial loot table
### ADDED: Tent to supermarket loot table
### ADDED: Range Finder to military loot table
### ADDED: Salute animation (key: , [COMMA])
### ADDED: Can give Morphine, Antibiotics and Epi-Pens to other players
### CHANGED: Completely new inventory UI (rightclick to use/equip items)
### CHANGED: Holding weapons will free up 10 slots, while carrying them takes 10
### CHANGED: Humanity Calculation
### CHANGED: Wood Piles from chopping trees are dropped on the ground
### CHANGED: Weapon damage tweaks
### CHANGED: Weapon magazine sizes tweaks
### CHANGED: Burst Fire for FN FAL
### CHANGED: DMR + Bizon PP-19 are now silent
### CHANGED: Decreased chance for wire fence to spawn
### CHANGED: Increased Bi-2 Airplane spawn chance to 35%
### CHANGED: Key for sitting is 'z'
### CHANGED: Patriot model is now an UAZ
### CHANGED: Volume is 0 when in a vehicle and engine is turned off
### REMOVED: Radiochat binding + radio chat itself
### REMOVED: Randomized damage (uses distance calculation now)
### FIXED: Not being able to access tents
### FIXED: Debug monitor appearing when opening inventory
### FIXED: Heroes bleeding when hit by a weapon that does less than 1000 damage
### FIXED: Using the wrong items in inventory once you have more than 11


# - #
## [0.9.4.3a] - 2015-08-12

[addons]/addon_gip/gip_c.lua
--[[
/**
	@ name: Give-an-Item Panel (GIP)
	@ author: Renkon
	@ version: 1.0
	@ type: DayZ Addon
	@ description: Panel which lets you give items to any player connected to your server.
*/
]]

addEvent("onGIPOpened", true)

local sW, sH = guiGetScreenSize() -- // Variables needed to know the width and height.
local w = 
{ 	
	gridList = {},
	label = {},
	editBox = {},
	button = {},
	comboBox = {}
}
	
local items = {
	["Weapons"] = {
		"M1911",
		"Revolver",
		"Makarov PM",
		"Bizon PP-19 SD",
		"G17",
		"M9",
		"Makarov SD",
		"PDW",
		"MP5A5",
		"Lee Enfield",
		"AK-74",
		"AKS-74U",
		"RPK",
		"AKM",
		"Sa58V CCO",
		"Sa58V RCO",
		"FN FAL",
		"M24",
		"DMR",
		"M40A3",
		"G36A CAMO",
		"G36C",
		"G36C CAMO",
		"G36K CAMO",
		"L85A2 RIS Holo",
		"M16A2",
		"M16A2 M203",
		"M4A1",
		"M16A4",
		"CZ 550",
		"SVD Dragunov",
		"Mosin-Nagant",
		"Winchester 1866",
		"Double-barreled Shotgun",
		"M1014",
		"Remington 870",
		"Compound Crossbow",
		"Baseball Bat",
		"Crowbar",
		"Hatchet",
		"Machete",
		"Parachute",
		"Grenade",
		"Hunting Knife",
		"Binoculars",
		"Range Finder",
		"Shovel",
		"Flashlight",
	},
	
	["Ammo"] = {
		"11.43x23mm Cartridge",
		"9x18mm Cartridge",
		"9x19mm Cartridge",
		".303 British Cartridge",
		"5.45x39mm Cartridge",
		"7.62x39mm Cartridge",
		"7.62x51mm Cartridge",
		"5.56x45mm Cartridge",
		".308 Winchester Cartridge",
		"7.62x54mm Cartridge",
		"1866 Slug",
		"12 Gauge Pellet",
		"Bolt",
	},
	
	["Food/Drinks"] = {
		"Baked Beans",
		"Pasta",
		"Sardines",
		"Frank & Beans",
		"Can (Corn)",
		"Can (Peas)",
		"Can (Pork)",
		"Can (Stew)",
		"Can (Ravioli)",
		"Can (Fruit)",
		"Can (Chowder)",
		"MRE",
		"Pistachios",
		"Trail Mix",
		"Water Bottle",
		"Soda Can (Pepsi)",
		"Soda Can (Cola)",
		"Soda Can (Mountain Dew)",
		"Can (Milk)",
		"Raw Meat",
		"Cooked Meat",
	},
	
	["Backpacks"] = {
		"Assault Pack (ACU)",
		"Czech Vest Pouch",
		"ALICE Pack",
		"Survival ACU",
		"British Assault Pack",
		"Backpack (Coyote)",
		"Czech Backpack",
	},
	
	["Toolbelt"] = {
		"Box of Matches",
		"IR Goggles",
		"NV Goggles",
		"GPS",
		"Map",
		"Toolbox",
		"Watch",
		"Radio Device",
		"Compass"
	},
	
	["Medical Items"] = {
		"Bandage",
		"Morphine",
		"Heat Pack",
		"Blood Bag",
		"Painkiller",
		"Antibiotics",
		"Epi-Pen",
		"Blood Bag (Empty)",
		"Blood Test Kit",
		"Blood Transfusion Kit",
	},
	
	["Car Parts"] = {
		"Tire",
		"Engine",
		"Tank Parts",
		"Main Rotary Parts",
		"Windscreen Glass",
		"Scrap Metal",
	},
	
	["Clothes"] = {
		"Survivor Clothing",
		"Civilian Clothing",
		"Camouflage Clothing",
		"Ghillie Suit",
	},
	
	["Blueprints"] = {
		"M4 Blueprint",
		"CZ 550 Blueprint",
		"Winchester 1866 Blueprint",
		"SPAZ-12 C. Shtgn. Blueprint",
		"Sawn-Off Shtgn. Blueprint",
		"AK-47 Blueprint",
		"Lee Enfield Blueprint",
		"Sporter 22 Blueprint",
		"Mosin 9130 Blueprint",
		"Crossbow Blueprint",
		"SKS Blueprint",
		"Blaze 95 D. R. Blueprint",
		"Remington 870 Blueprint",
		"FN FAL Blueprint",
		"G36C Blueprint",
		"Sa58V CCO Blueprint",
		"SVD Dragunov Blueprint",
		"DMR Blueprint",
		"M1911 Blueprint",
		"M9 SD Blueprint",
		"PDW Blueprint",
		"G17 Blueprint",
		"MP5A5 Blueprint",
		"Bizon PP-19 Blueprint",
		"Revolver Blueprint",
		"Desert Eagle Blueprint",
		"Hunting Knife Blueprint",
		"Hatchet Blueprint",
		"Baseball Bat Blueprint",
		"Shovel Blueprint",
		"Golf Club Blueprint",
		"Machete Blueprint",
		"Crowbar Blueprint",
		"Parachute Blueprint",
		"Tear Gas Blueprint",
		"Grenade Blueprint",
		"Binoculars Blueprint",
		".45 ACP Cartridge Blueprint",
		"9x19mm SD Cartridge Blueprint",
		"9x19mm Cartridge Blueprint",
		"9x18mm Cartridge Blueprint",
		"5.45x39mm Cartridge Blueprint",
		"5.56x45mm Cartridge Blueprint",
		"1866 Slug Blueprint",
		"2Rnd. Slug Blueprint",
		"12 Gauge Pellet Blueprint",
		"9.3x62mm Cartridge Blueprint",
		".303 British Cartridge Blueprint",
		"Bolt Blueprint",
		"Medic Kit Blueprint",
		"Wire Fence Blueprint",
		"Tent Blueprint",
		"Camouflage Clthng. Blueprint",
		"Survivor Clthng. Blueprint",
		"Civilian Clthng. Blueprint",
		"Ghillie Suit Blueprint",
		"Road Flare Blueprint",
		"Toolbox Blueprint",
		"Radio Device Blueprint",
		"IR Goggles Blueprint",
		"NV Goggles Blueprint",
	},
	
	["Blueprint Parts"] = {
		"Gun Barrel",
		"Short Gun Barrel",
		"Gun Stock",
		"Thread",
		"Cloth",
		"Gun Powder",
		"Mechanical Supplies",
		"Cables",
		"Nails",
		"Sheet",
		"Barbed Wire",
		"Duct Tape",
		"Glue",
		"Drugs",
		"Bandaid",
		"Vitamins",
		"Tissue",
		"Small Box",
		"String",
		"Needle",
		"Microchips",
		"Optics",
		"Sharp Metal",
		"Handle",
		"Wooden Stick",
		"Hand Saw",
		"Metal Plate",
		"Metallic Stick",
		"Small Casing",
		"DIY (Wood)",
		"DIY (Metal)",
		"Log",
		"Stone",
		"Plank",
		"RSJ",
		"Metal Sheet",
		"Metal Sheet (Rusted)",
	},
	
	["Other"] = {
		"Wood Pile",
		"Empty Gas Canister",
		"Full Gas Canister",
		"Road Flare",	
		"Wire Fence",
		"Tent",
		"San Fierro Carrier Keycard",
		"Area 69 Keycard"
	},
}

local isItemSelected = false

-- // Window creation code --
addEventHandler("onClientResourceStart", resourceRoot,
function()
	w.main = guiCreateWindow(sW/2 - 179, sH/2 - 165, 358, 330, "Give-an-Item Panel (GIP)", false)
	guiWindowSetSizable(w.main, false)
	guiSetAlpha(w.main, 1.00)

	w.gridList.main = guiCreateGridList(10, 26, 151, 291, false, w.main)
	w.gridList.column = guiGridListAddColumn( w.gridList.main, "Player", 0.85 )
	
	w.label.desc = guiCreateLabel(176, 26, 178, 22, "Choose a player from the gridlist!", false, w.main)
	guiSetFont(w.label.desc, "default-bold-small")
	w.label.desc2 = guiCreateLabel(233, 210, 60, 22, "Quantity", false, w.main)
	guiSetFont(w.label.desc2, "default-bold-small")

	w.comboBox.category = guiCreateComboBox(171, 65, 266, 160, "-- Choose a category --", false, w.main)
	w.comboBox.itemList = guiCreateComboBox(171, 145, 266, 50, "-- Choose an item --", false, w.main)
	
	w.editBox.quantity = guiCreateEdit(231, 234, 55, 26, "", false, w.main)
	w.button.give = guiCreateButton(176, 280, 77, 37, "Give", false, w.main)
	w.button.close = guiCreateButton(263, 280, 77, 37, "Close", false, w.main)
	guiSetFont(w.button.give, "default-bold-small")
	guiSetProperty(w.button.give, "NormalTextColour", "FFAAAAAA")
	guiSetFont(w.button.close, "default-bold-small")
	guiSetProperty(w.button.close, "NormalTextColour", "FFAAAAAA")
	
	guiSetVisible(w.main, false)
	guiSetEnabled(w.button.give, false)
	
	for key, value in pairs (items) do
		if type(value) == "table" then
			guiComboBoxAddItem(w.comboBox.category, key)
		end
	end
	
	-- // Making quantity be only edited by numbers --
	addEventHandler("onClientGUIChanged", w.editBox.quantity, 
	function()
		guiSetText(source, guiGetText(source):gsub("[^0-9]","")) -- // We remove everything which is not a number
		check()
	end)
	
	-- // Changing second combobox items when first one changes --
	addEventHandler("onClientGUIComboBoxAccepted", w.comboBox.category,
	function()
		local text = guiComboBoxGetItemText(w.comboBox.category, guiComboBoxGetSelected(w.comboBox.category))
		guiComboBoxClear(w.comboBox.itemList)
		for i, st in ipairs(items[text]) do
			guiComboBoxAddItem(w.comboBox.itemList, st)
		end
		guiComboBoxAdjustHeight(w.comboBox.itemList, #items[text])
	end )
	
	-- // Showing item when second one was accepted. --
	addEventHandler("onClientGUIComboBoxAccepted", w.comboBox.itemList,
	function()
		isItemSelected = true
		check()
	end )
	
	-- // Handling button that closes. --
	addEventHandler ( "onClientGUIClick", w.button.close, 
	function()
		guiSetVisible(w.main, false)
		showCursor(false)
		guiSetInputMode( "allow_binds" )
	end, false )
	
	-- // Handling give button. --
	addEventHandler ( "onClientGUIClick", w.button.give, 
	function()
		local playerName = guiGridListGetItemText(w.gridList.main, guiGridListGetSelectedItem(w.gridList.main), 1)
		local item = guiComboBoxGetItemText(w.comboBox.itemList, guiComboBoxGetSelected(w.comboBox.itemList))
		local giver = getPlayerName(localPlayer)
		if (getPlayerFromName(playerName)) then
			triggerServerEvent("onGIPGive", localPlayer, playerName, item, tonumber(guiGetText(w.editBox.quantity)))
		else
			outputChatBox("Player disconnected or changed name!", 255, 0, 0)
				end
		guiSetVisible(w.main, true)
		showCursor(true)
	end, false )

	addEventHandler("onClientGUIClick", w.gridList.main, check)
end )

-- // Remi-X function. Sets height depending on item quantity --
function guiComboBoxAdjustHeight ( combobox, itemcount )
    local width = guiGetSize ( combobox, false )
    return guiSetSize ( combobox, width, ( itemcount * 20 ) + 20, false )
end

-- // Handling when the administrator opens the panel --
addEventHandler("onGIPOpened", root,
function()
	guiGridListClear ( w.gridList.main )
	guiSetEnabled(w.button.give, false)  
	isItemSelected = false
	for id, player in ipairs(getElementsByType("player")) do
		if player ~= localplayer then
			local row = guiGridListAddRow ( w.gridList.main )
			guiGridListSetItemText ( w.gridList.main, row, w.gridList.column, getPlayerName ( player ), false, false )
		end
	end
	guiSetVisible(w.main, true)
	showCursor(true)
	guiSetInputMode( "no_binds_when_editing" )
end )

function check()
	if guiGetText(w.editBox.quantity) ~= "" and isItemSelected and guiGridListGetItemText(w.gridList.main, guiGridListGetSelectedItem(w.gridList.main), 1) ~= "" then -- // If there's quantity
		guiSetEnabled(w.button.give, true) -- // We enable the button
	end
end